### PR TITLE
Upgrade Android sample to Gradle Plugin 3, targeting API 26

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

**Comment by eskatos** [comment]
When testing, consider #584

File: samples/hello-android/settings.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 pluginManagement {
     repositories {
         gradlePluginPortal()
-        maven { url = uri("https://jcenter.bintray.com/") }
+        maven { url = uri("https://maven.google.com") }
```
Comment (jaredsburrows) [inline samples/hello-android/settings.gradle.kts]: Can you not use `google()` here? or atleast `google().url`?

File: samples/hello-android/build.gradle.kts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-    id("com.android.application") version "2.3.3"
+    id("com.android.application") version "3.0.0"
     kotlin("android") version "1.1.51"
 }
 
 android {
-    buildToolsVersion("25.0.0")
-    compileSdkVersion(23)
+    buildToolsVersion("27.0.0")
+    compileSdkVersion(26)
```
Comment (jaredsburrows) [inline samples/hello-android/build.gradle.kts]: You can target 27 for `compileSdkVersion` and `targetSdkVersion`.

File: doc/getting-started/Configuring-Plugins.md [hunk 3]
```diff
@@ -100,12 +100,12 @@ plugin extension accessors, in your `build.gradle.kts` file:
...
 ```kotlin
 plugins {
-    id("com.android.application") version "2.3.3"
+    id("com.android.application") version "3.0.0"
 }
 
 android {
-    buildToolsVersion("25.0.0")
-    compileSdkVersion(23)    
+    buildToolsVersion("27.0.2")
```
Comment (jaredsburrows) [inline doc/getting-started/Configuring-Plugins.md]: Is build tools `27.0.2` out yet?

**Comment by jaredsburrows** [comment]
Thanks for this pr!

File: samples/hello-android/settings.gradle.kts (hunk 1 above)
Comment (eskatos) [reply samples/hello-android/settings.gradle.kts]: No, not yet ;)

File: doc/getting-started/Configuring-Plugins.md (hunk 3 above)
Comment (eskatos) [reply doc/getting-started/Configuring-Plugins.md]: Yes, it is

File: samples/hello-android/build.gradle.kts (hunk 2 above)
Comment (eskatos) [reply samples/hello-android/build.gradle.kts]: Right. I purposely targeted API 26, aka Android 8, aka Oreo, as this is the latest released version. I suppose we could target API 27 already.

**Comment by eskatos** [comment]
Superseded by #586

**Pull request closed without merging**